pennylane/qchem/factorization.py

```diff
@@ -135,46 +214,192 @@ def factorize(two_electron, tol_factor=1.0e-5, tol_eigval=1.0e-5):
+
+        # Since core_tensors are symmetric and not contrained to be rank-one
+        upr_tri, unitary = jsp.linalg.schur(core_tensors)
+        factors = qml.math.array(
+            jnp.einsum(
+                "tpk,tqk,tki->tpqi",
+                leaf_tensors,
+                leaf_tensors,
+                unitary @ jnp.sqrt(upr_tri.astype(jnp.complex64)),
+            ),  # einsum contraction for them: "tpqi, trsi -> pqrs"
+            like=interface,
+        )
 
-    eigvals_r, eigvecs_r = np.linalg.eigh(two)
-    eigvals_r = np.array([val for val in eigvals_r if abs(val) > tol_factor])
+    return factors, core_tensors, leaf_tensors
 
-    eigvecs_r = eigvecs_r[:, -len(eigvals_r) :]
 
+def _double_factorization_eigen(two, tol_factor=1.0e-10, shape=None, interface=None):
+    """Double factorization via generalized eigen decomposition"""
+    eigvals_r, eigvecs_r = qml.math.linalg.eigh(two)
+    eigvals_r = qml.math.array([val for val in eigvals_r if abs(val) > tol_factor])
+
+    eigvecs_r = eigvecs_r[:, -len(eigvals_r) :]
     if eigvals_r.size == 0:
         raise ValueError(
             "All factors are discarded. Consider decreasing the first threshold error."
         )
+    vectors = eigvecs_r @ qml.math.diag(qml.math.sqrt(eigvals_r))
+
+    n, r = shape[0], len(eigvals_r)
+    factors = qml.math.array([vectors.reshape(n, n, r)[:, :, k] for k in range(r)], like=interface)
+    return factors
+
+
+def _double_factorization_cholesky(
+    two, tol_factor=1.0e-10, shape=None, interface=None, num_factors=None
+):
+    """Double factorization via Cholesky decomposition"""
+    n2 = shape[0] * shape[1]
+    if num_factors is None:
+        num_factors = n2
+
+    cholesky_vecs = qml.math.zeros((n2, num_factors), like=interface)
+    cholesky_diag = qml.math.array(qml.math.diagonal(two).real, like=interface)
+
+    for idx in range(num_factors):
+        if (max_err := qml.math.max(cholesky_diag)) < tol_factor:
+            cholesky_vecs = cholesky_vecs[:, :idx]
+            break
+
+        max_idx = qml.math.argmax(cholesky_diag)
+        cholesky_mat = cholesky_vecs[:, :idx]
+        cholesky_vec = (
+            two[:, max_idx] - cholesky_mat @ qml.math.conjugate(cholesky_mat[max_idx])
+        ) / qml.math.sqrt(max_err)
+
+        cholesky_vecs[:, idx] = cholesky_vec
+        cholesky_diag -= qml.math.abs(cholesky_vec) ** 2
+
+    factors = cholesky_vecs.T.reshape(-1, shape[0], shape[0])
+    return factors
+
+
+def _double_factorization_compressed(
+    two, optimizer, num_factors, num_steps=1000, init_params=None, prefactor=1e-5, norm_order=None
+):
+    """Compressed double factorization with optional regularization"""
+    norb = two.shape[0]
+    leaf_tensors, core_tensors = jnp.zeros((2, 0, norb, norb))
+
+    cost_func = value_and_grad(
+        partial(_compressed_cost_fn, two=two, norm_order=norm_order, prefactor=prefactor)
+    )
+
+    @jit
```

Comment:
How helpful this is for imp[roving performance?

Reply:
Quite a lot, minimum of 10x in the examples I was testing it with. Without `jit` computation will be too slow.